File: afc_saaspro_tax/afc_rest_apis/javascript/scripts/model/ProRatedTransaction.js
/**
 * Parameter for pro-rated tax calculation API.
 */

function ProRatedTransaction() {

    this.ProRatedPercent = 0;
    this.Transaction = null;

    /**
     * Sets the percentage to prorate specified as a decimal.
     * @param proRatedPercent Number
     * @return Reference to this object for method chaining. 
     */
    this.proRatedPercent = function (proRatedPercent) {
        this.ProRatedPercent = proRatedPercent;
        return this;
    }

    /**
     * Percentage to prorate specified as a decimal.
     * @return Number
     */
    this.getProRatedPercent = function () {
        return this.ProRatedPercent;
    }

    /* Reference to this object for method chaining.*
     * Sets the percentage to prorate specified as a decimal.
     * @param proRatedPercent Number 
     */
    this.setProRatedPercent = function (proRatedPercent) {
        this.ProRatedPercent = proRatedPercent;
    }

    /**
     * Sets the transaction to be processed.
     * @param transaction Transaction object.
     * @return Reference to this object for method chaining. 
     */
    this.transaction = function (transaction) {
        this.Transaction = transaction;
        return this;
    }

    /**
     * Transaction data.
     * @return Transaction object.
     */
    this.getTransaction = function () {
        return this.Transaction;
    }

    /* Reference to this object for method chaining.*
     * Sets the transaction to be processed.
     * @param transaction Transaction object. 
     */
    this.setTransaction = function (transaction) {
        this.Transaction = transaction;
    }
}

File: afc_saaspro_tax/afc_rest_apis/java/src/avalara/comms/rest/sample/model/ProRatedTransaction.java
package avalara.comms.rest.sample.model;

import com.google.gson.annotations.SerializedName;

/**
 * Parameter for pro-rated tax calculation API.
 */
public class ProRatedTransaction {

    @SerializedName("ProRatedPercent")
    private Double proRatedPercent = null;

    @SerializedName("Transaction")
    private Transaction transaction = null;

    /**
     * Sets the percentage to prorate specified as a decimal.
     * @param proRatedPercent
     * @return 
     */
    public ProRatedTransaction proRatedPercent(Double proRatedPercent) {
        this.proRatedPercent = proRatedPercent;
        return this;
    }

    /**
     * Percentage to prorate specified as a decimal.
     * @return proRatedPercent
     */
    public Double getProRatedPercent() {
        return proRatedPercent;
    }

    /**
     * Sets the percentage to prorate specified as a decimal.
     * @param proRatedPercent 
     */
    public void setProRatedPercent(Double proRatedPercent) {
        this.proRatedPercent = proRatedPercent;
    }

    /**
     * Sets the transaction to be processed.
     * @param transaction
     * @return 
     */
    public ProRatedTransaction transaction(Transaction transaction) {
        this.transaction = transaction;
        return this;
    }

    /**
     * Transaction data.
     * @return transaction
     */
    public Transaction getTransaction() {
        return transaction;
    }

    /**
     * Sets the transaction to be processed.
     * @param transaction 
     */
    public void setTransaction(Transaction transaction) {
        this.transaction = transaction;
    }
}
